pkg/collector/collector.go

```diff
@@ -1584,6 +1587,11 @@ func (c *Collector) registerWithSamplers(ctx context.Context) {
 
 				readyPct := (nodesWithResults.Len() * 100 / nodesWithSamplers.Len())
 				if nodesWithSamplers.Len() > 0 && readyPct > c.UtilizationServer.MinResultPctBeforeReady {
+					cyclesCount++
+					if cyclesCount > c.UtilizationServer.WaitSamplerRegistrationsBeforeServe {
+						c.UtilizationServer.ServeResults.Store(true)
+					}
+
```

Comment:
Not part of this PR.
To avoid "repetition." you can consider: 
```suggestion
log := log.WithValues("running-minutes", time.Since(c.startTime).Minutes(),
						"nodes-with-results-count", nodesWithResults.Len(),
						"nodes-with-samplers-count", nodesWithSamplers.Len(),
						"nodes-with-running-samplers-count", nodesWithRunningSamplers.Len(),
						"nodes-missing-results", nodesMissingResults.List(),
						"ready-pct", readyPct,
						"min-ready-pct", c.UtilizationServer.MinResultPctBeforeReady,
						"remaining-cycles", waitReadyCyclesCount)
```
with following: 
- `log.Info("collector ready")`
- `log.Info("collector has required sampler results, waiting on WaitSamplerRegistrationsBeforeReady to be ready")`
- `log.Info("collector not-ready")`

Note: `log := log.WithValues` will only be applicable in this closure.